ggshield/scan/scannable.py

```diff
@@ -127,6 +129,13 @@ def extra_headers(self) -> Dict[str, str]:
             **extra_headers,
         }
 
+    def get_tar_stream(self) -> BytesIO:
+        tar_stream = BytesIO()
```

Comment:
This is going to create the whole archive in memory, are we sure this won't be a problem? An alternative would be to pass an opened file object in which the method would write. This way it's up to the caller to decide if the archive should be created in memory or not.

Reply:
Discussed with Henri, one of the reasonings behind putting it in memory is also that it safeguards us as we will put it in memory on our side. One thing we will add in a next issue though is a mechanism to prevent creating archives which are too big

Reply:
OK. If we really want to generate the archive in memory, then there is no reason to return a BytesIO: we might as well rename this method `create_tar()` and make it return `bytes`.